src/router.rs

```diff
@@ -14,6 +16,27 @@ pub struct Router<'a> {
     schema: &'a Schema,
     matchers: BTreeMap<MatcherKey, Expression>,
     pub fields: HashMap<String, usize>,
+    regex_cache: HashMap<String, Rc<Regex>>,
+}
+
+fn release_cache(expr: &Expression, router: &mut Router) {
+    match expr {
+        Expression::Logical(l) => match l.as_ref() {
+            LogicalExpression::And(l, r) | LogicalExpression::Or(l, r) => {
+                release_cache(l, router);
+                release_cache(r, router);
+            }
+            LogicalExpression::Not(r) => release_cache(r, router),
+        },
+        Expression::Predicate(p) => {
+            if let Value::Regex(rc) = &p.rhs {
+                if Rc::strong_count(&rc) == 2 {
```

Comment:
```suggestion
                if Rc::strong_count(rc) == 2 {
```

<img width="797" alt="截屏2024-10-08 10 08 37" src="https://github.com/user-attachments/assets/cc495da2-1d42-4019-9f5d-cd1cbb5a61d3">

Reply:
Fixed! Thanks ✅